tests/tripal_chado/fields/ChadoFieldGetValuesListTest.php

```diff
@@ -0,0 +1,442 @@
+      )
+    );
+
+    // Ensure there are no more then 5 as specified in the limit above.
+    $this->assertLessThanOrEqual(5, sizeof($values),
+      t('Returned too many results for @field_name.', array('@field_name' => $field_name)));
+
+    // Ensure a known value is in the list.
+    // Note: The following generates fake data with a fixed value for the column this
+    // field is based on. This allows us to check that fixed value is one of those
+    // returned by ChadoField->getValueList().
+    $fake_value = $this->generateFakeData($info['bundle_base_table'], $info['base_schema'], $info['instance_info']['settings']['chado_column']);
+    if ($fake_value !== FALSE) {
+
+      // Re-generate the values...
+      $values = $instance->getValueList(array('limit' => 200));
+
+      // And ensure our fake value is in the returned list.
+      // We can only check this if all the results are returned.
+      // As such, we set the limit quite high above and if we have
+      // less then the limit, we will go ahead with the test.
+      // @note: this tests all fields on TravisCI since there is no pre-existing data.
+      if (sizeof($values) < 200) {
+        $this->assertContains($fake_value, $values, "\nThe following array should but does not contain our fake value ('$fake_value'): '" . implode("', '", $values) . '.');
+      }
+    }
+
+  }
+
+  /**
+   * DataProvider: a list of fields who store their data in the base table of a bundle.
+   *
+   * Each element describes a field instance and consists of:
+   *   - the machine name of the field (e.g. obi__organism).
+   *   - the machine name of the bundle (e.g. bio_data_17).
+   *   - an array of additional information including:
+   *       - instance_info: information about the field instance.
+   *       - field_info: information about the field.
+   *       - bundle: the TripalBundle object.
+   *       - bundle_base_table: if applicable, the chado base table the bundle stores it's data in.
+   *       - base_schema: the Tripal Schema array for the bundle_base_table.
+   */
+  public function getBaseFields() {
+
+    // Retrieve a list of fields to test.
+    // Note: this list is cached to improve performance.
+    $fields = $this->retrieveFieldList();
+
+    return $fields['field_chado_storage']['base'];
+  }
+
+  /**
+   * Test for fields based on columns in the base table that are also foreign keys.
+   *
+   * @dataProvider getBaseFkFields
+   * @group current
+   * @group fields
+   * @group getValueList
+   */
+  public function testBaseTableForeignKey($field_name, $bundle_name, $info) {
+    include_once(drupal_get_path('tripal_chado', 'module') . '/includes/TripalFields/ChadoField.inc');
+
+    // Construct the Field instance we want the values for.
+    // Specifying "ChadoField" here ensures we are only testing our
+    // implementation of getValueList() and not the custom version for any
+    // given field.
+    // YOU SHOULD TEST CUSTOM FIELD IMPLEMENTATIONS SEPARATELY.
```

Comment:
im seeing this method as a super helpful way to test custom fields.  Maybe a new section in the "creating custom fields" guide?